x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/IdentifierBuilder.java

```diff
@@ -48,10 +53,32 @@ public String visitIndexPattern(List<EsqlBaseParser.IndexPatternContext> ctx) {
         List<String> patterns = new ArrayList<>(ctx.size());
         ctx.forEach(c -> {
             String indexPattern = visitIndexString(c.indexString());
+            validateIndexPattern(indexPattern, c);
```

Comment:
How about the `c.clusterString()`? Don't we want to validate that?

Reply:
> How about the `c.clusterString()`? Don't we want to validate that?

The validation of cluster alias is out of scope :-), but let me see if I can find a good way to validate it, if it is not very complicated, I'll add it. Otherwise we can do it in a separate PR.

Reply:
I looked into the `clusterString` a bit more, it seems like we don't have as much restrictions on a cluster string as on index names. I prefer to skip the validation of `clusterString`, as there is not much to validate :), and it should not add too much overhead on data node, if it is wrong.

Reply:
> we don't have as much restrictions on a cluster string

Fair. It might be that the grammar does that for us already.